--- dhcpcd-10.2.4.orig/compat/arc4random.h	2025-09-22 16:35:08.000000000 +0000
+++ dhcpcd-10.2.4/compat/arc4random.h	2025-09-22 16:35:28.000000000 +0000
@@ -11,6 +11,9 @@
 #define ARC4RANDOM_H
 
 #include <stdint.h>
+#ifdef __QNX__
+#include <stddef.h>
+#endif
 
 uint32_t arc4random(void);
 void arc4random_buf(void *, size_t);
--- dhcpcd-10.2.4.orig/compat/setproctitle.h	2025-09-22 16:35:08.000000000 +0000
+++ dhcpcd-10.2.4/compat/setproctitle.h	2025-09-22 16:35:28.000000000 +0000
@@ -33,6 +33,7 @@
 #endif
 #endif /* !__printflike */
 
+#ifndef __QNX__
 /* WEXITSTATUS is defined in stdlib.h which defines free() */
 #ifdef WEXITSTATUS
 static inline const char *
@@ -46,6 +47,7 @@
 	free(name);
 }
 #endif
+#endif // __QNX__
 
 void setproctitle_init(int, char *[], char *[]);
 __printflike(1, 2) void setproctitle(const char *, ...);
--- dhcpcd-10.2.4.orig/configure	2025-09-22 16:35:08.000000000 +0000
+++ dhcpcd-10.2.4/configure	2025-09-22 16:35:28.000000000 +0000
@@ -370,9 +370,10 @@
 	ALLOW_USR_LIBS=true
 fi
 case "$OS" in
-linux*|solaris*|sunos*|kfreebsd*|dragonfly*|freebsd*) ;;
+linux*|solaris*|sunos*|kfreebsd*|dragonfly*|freebsd*|qnx*) ;;
 *)
-	# There might be more than one ...
+	echo "$OS"
+        # There might be more than one ...
 	for LDELFN in /libexec/ld-elf.so.[0-9]*; do
 		[ -x "$LDELFN" ] && break
 	done
